static/js/components/AuthenticationAlert.test.tsx

```diff
@@ -54,28 +54,27 @@ describe("Prompting for authentication", () => {
   ])(
     "prompts for authentication when APIs reject with auth errors",
     async () => {
-      const { result, setMockWebsiteResponse, website, user } = await setup()
-
-      setMockWebsiteResponse(authRejectionBody, 403)
+      await withFakeLocation(async () => {
+        const { result, setMockWebsiteResponse, website, user } = await setup()
 
-      const siteLink = await waitFor(() => result.getByText(website.title))
-      await act(() => user.click(siteLink))
+        setMockWebsiteResponse(authRejectionBody, 403)
 
-      const dialog = await waitFor(() => result.getByRole("dialog"))
+        const siteLink = await waitFor(() => result.getByText(website.title))
+        await act(() => user.click(siteLink))
 
-      expect(dialog).toHaveTextContent("Session Expired")
-      expect(dialog).toHaveTextContent("Please log in and try again.")
+        const dialog = await waitFor(() => result.getByRole("dialog"))
 
-      await withFakeLocation(async () => {
+        expect(dialog).toHaveTextContent("Session Expired")
+        expect(dialog).toHaveTextContent("Please log in and try again.")
         const goToLogin = dom.queryByText(dialog, "Go to Login")
         assertInstanceOf(goToLogin, HTMLButtonElement)
         await act(() => user.click(goToLogin))
-        await waitFor(() => {
-          expect(window.location.href).toBe("/login/saml/?idp=default")
-        })
+        expect(window.location.href).toBe("/login/saml/?idp=default")
+        result.unmount()
       })
 
-      result.unmount()
+      // Test is inside a callback. Let's make sure it actually ran.
+      expect.assertions(3)
```

Comment:
Previously, the callback inside `withFakeLocation` was never actually being run.

Now it is being run, and we are sure of that because of `expect.assertions(3)`.